src/array_api_stubs/_draft/searching_functions.py

```diff
@@ -87,6 +87,35 @@ def nonzero(x: array, /) -> Tuple[array, ...]:
     """
 
 
+def searchsorted(
+    x1: array,
+    x2: array,
+    /,
+    *,
+    side: Literal["left", "right"] = "left",
+    sorter: Optional[array] = None,
+) -> array:
+    """
+    Finds the indices into ``x1`` such that, if the corresponding elements in ``x2`` were inserted before the indices, the order of ``x1``, when sorted in ascending order, would be preserved.
+
+    Parameters
+    ----------
+    x1: array
+        input array. Must be a one-dimensional array. If ``sorter`` is ``None``, must be sorted in ascending order; otherwise, ``sorter`` must be an array of indices that sort ``x1`` in ascending order.
+    x2: array
+        array containing search values.
```

Comment:
Any reason why this doesn't say that `x2` must be a one-dimensional array?

Reply:
Because `x2` is [allowed](https://numpy.org/doc/stable/reference/generated/numpy.searchsorted.html) to be multi-dimensional. Only `x1` is required to be one-dimensional. The output array shape should be the same as `x2`. I don't see a reason why we'd need to constrain `x2` to be one-dimensional.

Reply:
Okay, the proposal in gh-688 said 1-D, but this seems right. I checked that all of NumPy, CuPy, JAX, PyTorch and Dask support n-D arrays here - resolving.